### PR TITLE
:hammer: Fixes #480 material property type

### LINKED ISSUE #480: Material property test for bool
Function `property` in `material.h` is only outputting `double` material properties.
It does not work when we have a `bool` material property such as `softening` in Mohr-Coulomb and what I am developing with bonded NorSand.

[Code Line](https://github.com/cb-geo/mpm/blob/164a6665a2b791f893e0e5492c44625acda36de7/include/material/material.h#L59)

Suggestion:
Overloading the function with `bool`?

### PR DESCRIPTION
**Pull request by kks32**
call property as `material->template property<type>(key)`

### CODE REVIEW

File: include/cell.tcc [hunk 1]
```diff
@@ -510,15 +510,21 @@ inline Eigen::Matrix<double, 3, 1> mpm::Cell<3>::local_coordinates_point(
   return xi;
 }
 
+//! Return the local coordinates of a point in a 1D cell
+template <>
+inline Eigen::Matrix<double, 1, 1> mpm::Cell<1>::transform_real_to_unit_cell(
```
Comment (ezrayst) [inline include/cell.tcc:515]: Why do we have this function here? is it from #479 that has not been accounted for?

File: tests/material/bingham_test.cc [hunk 2]
```diff
@@ -69,11 +69,11 @@ TEST_CASE("Bingham is checked in 2D", "[material][bingham][2D]") {
     REQUIRE(material->id() == 0);
 
     // Get material properties
-    REQUIRE(material->property("density") ==
+    REQUIRE(material->template property<double>("density") ==
```
Comment (ezrayst) [inline tests/material/bingham_test.cc:72]: Great!

File: include/cell.tcc (hunk 1 above)
Comment (kks32) [reply include/cell.tcc:515]: clang failed, at least locally on my machine

Comment (ezrayst) [reply include/cell.tcc:515]: Understood, it's fine then!

**Review by ezrayst — APPROVED** [review]

**Comment by ezrayst** [comment]
Codecov decreases though :(

**Pull request merged**